### PR TITLE
alot: 0.9.1 -> 0.10

### PR DESCRIPTION
**Pull request by lucc**
###### Motivation for this change

New release of alot: https://github.com/pazz/alot/releases/tag/0.10

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@edibopp please review.

### CODE REVIEW

**Review by fabaff — COMMENTED** [review]
Please update the commit message. It's located in the Python modules area.

```bash
python3Packages.alot: 0.9.1 -> 0.10
```

Could you also update the license please?  It's `gpl3Plus` (-> https://github.com/pazz/alot/blob/master/setup.py#L22 and the file headers).

**Comment by lucc** [comment]
@fabaff fixed.

Does anybody know how I can exclude certain unittests? alot uses `python setup.py test` with the standard lib unitests module from python.  There are exactly two tests that fail in NixOS' sandbox:
```
======================================================================
FAIL: test_no_spawn_no_stdin_attached (tests.commands.test_global.TestExternalCommand)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/tests/utilities.py", line 189, in _actual
    return loop.run_until_complete(coro(*args, **kwargs))
  File "/nix/store/i1m8r7mv8h47wr850cdsxksy22lv6gsz-python3-3.9.6/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/build/source/tests/commands/test_global.py", line 129, in test_no_spawn_no_stdin_attached
    ui.notify.assert_not_called()
  File "/nix/store/i1m8r7mv8h47wr850cdsxksy22lv6gsz-python3-3.9.6/lib/python3.9/unittest/mock.py", line 868, in assert_not_called
    raise AssertionError(msg)
AssertionError: Expected 'notify' to not have been called. Called 1 times.
Calls: [call('editor has exited with error code 1 -- No stderr output', priority='error')].

======================================================================
FAIL: test_env_set (tests.test_helper.TestCallCmdAsync)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/tests/utilities.py", line 189, in _actual
    return loop.run_until_complete(coro(*args, **kwargs))
  File "/nix/store/i1m8r7mv8h47wr850cdsxksy22lv6gsz-python3-3.9.6/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/build/source/tests/test_helper.py", line 433, in test_env_set
    self.assertEqual(ret[0], 'bar')
AssertionError: '' != 'bar'
+ bar
```
 and because of this we have `doCheck=false` in the alot derivation.  I would like to deactivate these two and run the remaining tests.

**Comment by lucc** [comment]
I marked the two problematic tests as expected failure. This is now ready for review (and merge from my point of view).

**Comment by Artturin** [comment]
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pytestcheckhook-using-pytestcheckhook

**Comment by lucc** [comment]
@Artturin thanks for the hint that is much nicer.

File: pkgs/development/python-modules/alot/default.nix [hunk 1]
```diff
@@ -35,11 +35,18 @@ buildPythonPackage rec {
     gpgme
   ];
 
-  # some twisted tests need the network (test_env_set... )
-  doCheck = false;
   postBuild = lib.optionalString withManpage "make -C docs man";
 
-  checkInputs =  [ gawk future mock gnupg procps ];
+  doCheck = true;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/alot/default.nix:40]:
```suggestion
```
default

File: pkgs/development/python-modules/alot/default.nix [hunk 2]
```diff
@@ -35,11 +35,18 @@ buildPythonPackage rec {
...
+  checkInputs = [ gawk future mock gnupg procps pytestCheckHook ];
+  #checkPhase = "pytest";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/alot/default.nix:42]:
```suggestion
  checkInputs = [ gawk future mock gnupg procps pytestCheckHook ];

```

File: pkgs/development/python-modules/alot/default.nix [hunk 3]
```diff
@@ -35,11 +35,18 @@ buildPythonPackage rec {
...
+  # some twisted tests need the network (test_env_set... )
+  disabledTests = [
+    "test_env_set"
+    "test_no_spawn_no_stdin_attached"
+     #"tests/test_helper.py::TestCallCmdAsync::test_env_set"
+     #"tests/commands/test_global.py::TestExternalCommand::test_no_spawn_no_stdin_attached"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/alot/default.nix:48]:
```suggestion
```

File: pkgs/development/python-modules/alot/default.nix [hunk 4]
```diff
@@ -35,11 +35,18 @@ buildPythonPackage rec {
...
 
-  # some twisted tests need the network (test_env_set... )
-  doCheck = false;
   postBuild = lib.optionalString withManpage "make -C docs man";
 
-  checkInputs =  [ gawk future mock gnupg procps ];
+  doCheck = true;
+  checkInputs = [ gawk future mock gnupg procps pytestCheckHook ];
+  #checkPhase = "pytest";
+  # some twisted tests need the network (test_env_set... )
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/alot/default.nix:43]:
```suggestion
  # some twisted tests need internet access
```

**Comment by lucc** [comment]
@SuperSandro2000 resolved.

**Pull request merged**